test/fhirtypes/ElementDefinition.constrainType.test.ts

```diff
@@ -1207,102 +1207,288 @@ describe('ElementDefinition R5', () => {
-      const onlyRule = new OnlyRule('activity.performedActivity');
-      onlyRule.types = [
-        { type: 'http://hl7.org/fhir/StructureDefinition/actualgroup', isReference: true }
-      ];
-      performedActivity.constrainType(onlyRule, fisher);
-      expect(performedActivity.type).toHaveLength(1);
-      expect(performedActivity.type[0]).toEqual(
-        new ElementDefinitionType('CodeableReference').withTargetProfiles(
-          'http://hl7.org/fhir/StructureDefinition/actualgroup'
-        )
-      );
-    });
-
-    it('should allow a CodeableReference to multiple resource types to be constrained to a reference to a single profile', () => {
-      const performedActivity = r5CarePlan.elements.find(
-        e => e.id === 'CarePlan.activity.performedActivity'
-      );
-      const onlyRule = new OnlyRule('activity.performedActivity');
-      onlyRule.types = [
-        { type: 'http://hl7.org/fhir/StructureDefinition/bodyheight', isReference: true },
-        { type: 'http://hl7.org/fhir/StructureDefinition/bodyweight', isReference: true }
-      ];
-      performedActivity.constrainType(onlyRule, fisher);
-      expect(performedActivity.type).toHaveLength(1);
-      expect(performedActivity.type[0]).toEqual(
-        new ElementDefinitionType('CodeableReference').withTargetProfiles(
-          'http://hl7.org/fhir/StructureDefinition/bodyheight',
-          'http://hl7.org/fhir/StructureDefinition/bodyweight'
-        )
-      );
-    });
-
-    it('should constrain an element to Reference type with the Reference() keyword when both Reference and CodeableReference is allowed', () => {
-      // R5 Extension allows both Reference and CodeableReference types
-      const r5Extension = fisher.fishForStructureDefinition('Extension');
-      const valueX = r5Extension.elements.find(e => e.id === 'Extension.value[x]');
-      // * value[x] only Reference(Resource)
-      const onlyRule = new OnlyRule('value[x]');
-      onlyRule.types = [{ type: 'Resource', isReference: true }];
-      valueX.constrainType(onlyRule, fisher);
-      expect(valueX.type).toHaveLength(1);
-      expect(valueX.type[0]).toEqual(
-        new ElementDefinitionType('Reference').withTargetProfiles(
-          'http://hl7.org/fhir/StructureDefinition/Resource'
-        )
-      );
-    });
-
-    it('should throw InvalidTypeError when a passed in reference to a type that cannot constrain any existing references to types on a CodeableReference', () => {
-      const addresses = r5CarePlan.elements.find(e => e.id === 'CarePlan.addresses');
-      const onlyRule = new OnlyRule('addresses');
-      onlyRule.types = [{ type: 'Patient', isReference: true }];
-      expect(() => {
-        addresses.constrainType(onlyRule, fisher);
-      }).toThrow(/"Reference\(Patient\).*Reference\(.*Condition\)/);
+    describe('Reference() keyword', () => {
+      it('should allow a CodeableReference to multiple resource types to be constrained to a reference to a subset', () => {
```

Comment:
I'd suggest rewording the tests that use Reference to constrain a CodeableConcept so that the test summary indicates that a warning should be logged. E.g.:
```suggestion
      it('should warn when a CodeableReference to multiple resource types is constrained to a Reference to a subset', () => {
```

Reply:
Updated in [9f88612](https://github.com/FHIR/sushi/pull/1292/commits/9f88612248639230232f7cb9157c6dd3bb37663b), along with the few other tests that needed this update.